packages/http.go

```diff
@@ -39,18 +38,27 @@ func ServePackage(w http.ResponseWriter, r *http.Request, p *Package) {
 		})
 		if err != nil {
 			logger.Error("archiving package path failed", zap.Error(err))
-			return
 		}
-	} else {
-		http.ServeFile(w, r, packagePath)
+		return
 	}
+
+	stream, err := os.Open(packagePath)
+	if err != nil {
+		logger.Error("failed to open file", zap.Error(err))
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+	defer stream.Close()
+
+	http.ServeContent(w, r, packagePath, f.ModTime(), stream)
```

Comment:
Do we still need these changes now that we are doing redirects?

Reply:
Well, in my opinion, it's much clearer and readable now.

Reply:
Agree with the increased cleanness after avoiding the `else`. But this is changing from `ServeFile` to `Open` + `ServeContent`. Is this change needed? Are we sure this is not going to change any behaviour?